### PR TITLE
Multi chrom analysis

### PR DESCRIPTION
**Pull request by jgallowa07**
multiple chromosomes at the top of the `Snakefile`  now all feed into a single `msmc` run. Currently working on doing this for `smc++` and `stairwayplot`

### CODE REVIEW

**Comment by jgallowa07** [comment]
p.s. this means you no longer get results for each target from every _combination_ of `reps` and `chromosomes`. Rather, the specified chromosomes will all be fed into a single run of each method. Finishing up on `stairwayplot` now.

File: n_t/Snakefile [hunk 1]
```diff
@@ -155,39 +152,58 @@ rule homo_sapiens_compound_smcpp:
...
+
+
 rule homo_sapiens_msmc:
     input:
-        "homo_sapiens_Gutenkunst/{seeds}.{chrms}.trees"
-    output:
-        "homo_sapiens_Gutenkunst/{seeds}.{chrms}.trees.multihep.txt.final.txt"
+        #TODO make this not dependent on all simulations. 
+        # seed = output.seed?
+        expand("homo_sapiens_Gutenkunst/{seeds}.{chrms}.trees.multihep.txt", \
```
Comment (jeromekelleher) [inline n_t/Snakefile]: The line breaking is a bit inconsistent here. Do we really need the backslash? (If it was Python code we wouldn't). The second line should definitely be indented though, it's very confusing having a broken line at the same indentation level.

**Comment by jgallowa07** [comment]
@jeromekelleher I just finished stairwayplot multi-chrom and am doing code cleanup now that I've tested it; probably should've waited on the PR but I put it up so others could test it out if they wanted. Will push again soon once it's cleaned up.

**Comment by andrewkern** [comment]
i'll take a look at this code in a few hours.

**Comment by jgallowa07** [comment]
should be cleaned up now @jeromekelleher @andrewkern if you would like to review. As soon as this gets merged I'll work on the generalization and file output structure.

**Comment by andrewkern** [comment]
I pulled this PR into it's own branch and am letting it run now. Should be a couple of hours.

File: n_t/msmc.py [hunk 2]
```diff
@@ -22,14 +22,16 @@ def write_msmc_file(path):
     for var in ts.variants():
         cur = int(var.site.position)
         if cur > prev:
-            string = chrom+'\t'+str(cur)+'\t'+str(cur-prev)+'\t'+''.join(map(str,var.genotypes))+"\n"
-            fi.write(string)
+            #string = chrom+'\t'+str(cur)+'\t'+ str(cur-prev)
+            #string_2 = '\t'+''.join(map(str,var.genotypes))+"\n"
+            geno = ''.join(map(str,var.genotypes))
+            fi.write(f"{chrom}\t{cur}\t{cur-prev}\t{geno}\n")
```
Comment (andrewkern) [inline n_t/msmc.py]: @jgallowa07 msmc can only deal with small samples, and smc++ and stairwayplot both can use larger sample sizes. I think we want to subsample our original tree sequences, choosing some specified number of individuals at random, and use that subsample for msmc. Can you please adjust this function to do that?

Comment (jgallowa07) [reply n_t/msmc.py]: totally, np.

File: n_t/msmc.py [hunk 3]
```diff
@@ -22,14 +22,16 @@ def write_msmc_file(path):
     for var in ts.variants():
         cur = int(var.site.position)
         if cur > prev:
-            string = chrom+'\t'+str(cur)+'\t'+str(cur-prev)+'\t'+''.join(map(str,var.genotypes))+"\n"
-            fi.write(string)
+            #string = chrom+'\t'+str(cur)+'\t'+ str(cur-prev)
```
Comment (jeromekelleher) [inline n_t/msmc.py]: No point in keeping this lying around, delete.

Comment (jgallowa07) [reply n_t/msmc.py]: Gotcha; I have more comments that need deleted too. I'll squash and push once I finish these changes for Andy; Should be soon.

**Comment by andrewkern** [comment]
lgtm. ran fine for me too-- great work here @jgallowa07. i'm going to merge now.

**Pull request merged**